xonsh/environ.py

```diff
@@ -1647,6 +1655,88 @@ def _repr_pretty_(self, p, cycle):
                 p.break_()
                 p.pretty(dict(self))
 
+    def register(self, varname, vartype=None, defaultval=None, vardoc=None,
+                 ens_validate=None, ens_convert=None, ens_detype=None,
+                 doc_configurable=None, doc_default=None, doc_store_as_str=None):
+        """Register an enviornment variable with optional type handling,
+        default value, doc.
+
+        Parameters
+        ----------
+        varname : str
+            Environment variable name to register. Typically all caps.
+        vartype : str, optional,  {'bool', 'str', 'path', 'int', 'float'}
+            Variable type. If not one of the available presets, use `ens_validate`,
+            `ens_convert`, and `ens_detype` to specify type behavior.
+        defaultval : optional
+            Default value for variable. ``ValueError`` raised if type does not match
+            that specified by `vartype` (or `ens_validate`).
+        vardoc : str, optional
+            Docstring for variable.
+        ens_validate : func, optional
+            Function to validate type.
+        ens_convert : func, optional
+            Function to convert variable from a string representation to its type.
+        ens_detype : func, optional
```

Comment:
These could probably drop the `ens_` prefix

Reply:
Done!